app/models/host.rb

```diff
@@ -1208,6 +1208,53 @@ def refresh_firewall_rules
   def refresh_advanced_settings
   end
 
+  def self.network_port_type
+    raise NotImplementedError, "must be implemented in a subclass"
+  end
+
+  def refresh_network_interfaces(ssu)
+    smartstate_network_ports = MiqLinux::Utils.parse_network_interface_list(ssu.shell_exec("ip a"))
+
+    neutron_network_ports = network_ports.where(:source => :refresh).each_with_object({}) do |network_port, obj|
```

Comment:
Hrm, I think I'm wrong ... it's starting to look like this method will only really work for OpenStack hosts.

Can you push this method down to OpenStack Host?  Then, maybe this implementation simply does nothing... In fact it might be nice to wrap this with a `supports feature` to keep non-OpenStack hosts from even trying to call it.  And, by default, `refresh_network_interfaces` would be unsupported for other providers.